controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    refboard_joint_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

refboard_joint_controller:
  ros__parameters:
    joints:
      - joint_x
      - joint_y
      - joint_z
      - joint_a
      - joint_b
      - joint_c
    
    command_interfaces:
      - position
    
    state_interfaces:
      - position
      - velocity
    
    action_monitor_rate: 20.0
    
    allow_partial_joints_goal: true
    open_loop_control: false
    allow_integration_in_goal_trajectories: true
    
    constraints:
      stopped_velocity_tolerance: 0.02
      goal_time: 0.0
      joint_x:
        trajectory: 0.05
        goal: 0.02
      joint_y:
        trajectory: 0.05
        goal: 0.02
      joint_z:
        trajectory: 0.05
        goal: 0.02
      joint_a:
        trajectory: 0.05
        goal: 0.02
      joint_b:
        trajectory: 0.05
        goal: 0.02
      joint_c:
        trajectory: 0.05
        goal: 0.02

joint_state_broadcaster:
  ros__parameters: {}
